pyccel/codegen/printing/ccudacode.py

```diff
@@ -519,19 +519,23 @@ def copy_CudaArray_Data(self, expr):
         declare_dtype = self.find_in_dtype_registry(self._print(rhs.dtype), rhs.precision)
         dtype = self.find_in_ndarray_type_registry(self._print(rhs.dtype), rhs.precision)
         arg = rhs.arg if isinstance(rhs, (CudaArray, CupyArray)) else rhs
+        if rhs.current_location == 'device' and rhs.memory_location == 'host':
+            errors.report("Cannot copy data from device to host", symbol=expr, severity='error')
```

Comment:
you can copy data from `device` to `host`.

the problem is when creating a `host` array from the `device`.

Reply:
Roger that i'll change that error message.

Reply:
Fixed in [aa5e773](https://github.com/pyccel/pyccel/pull/1228/commits/aa5e773c542e0ecc6f193ac775b43dfde1d35b16).